src/main/java/seedu/address/logic/commands/applicant/SortApplicant.java

```diff
@@ -0,0 +1,109 @@
+package seedu.address.logic.commands.applicant;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.Comparator;
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.applicant.Applicant;
+
+public class SortApplicant extends Command {
+
+    public static final String SORTAPPLICANT_BY_DATEAPPLIED = "dateapplied";
+
+    public static final String SORTAPPLICANT_BY_INTERVIEW = "interview";
+
+    public static final String SORTAPPLICANT_BY_JOB = "job";
+
+    public static final String COMMAND_WORD = "sortapplicant";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sorts all applicants by a given attribute.";
+
+    public static final String MESSAGE_SORT_APPLICANT_SUCCESS = "Applicants successfully sorted by %1$s.";
+
+    public static final String MESSAGE_CONSTRAINTS = "Applicants should be sorted by their "
+            + "'dateapplied', 'interview', or 'job'.";
+
+    public static final String ATTRIBUTE_DATEAPPLIED = "date applied";
+
+    public static final String ATTRIBUTE_INTERVIEW = "interview date";
+
+    public static final String ATTRIBUTE_JOB = "job applied";
+
+    public static final Comparator<Applicant> COMPARE_BY_DATEAPPLIED =
+            Comparator.comparing(Applicant::getDateApplied);
+
+    public static final Comparator<Applicant> COMPARE_BY_INTERVIEWDATE =
+            Comparator.comparing(Applicant::getInterviewDate);
+
+    public static final Comparator<Applicant> COMPARE_BY_JOBID =
+            Comparator.comparing(Applicant::getJobId);
+
+    private final Comparator<Applicant> compareBy;
+    private final String attribute;
+
+    /**
+     * Instantiates a new SortApplicant to sort the Applicant list by a specified attribute.
+     *
+     * @param compareBy the compare by
+     */
+    public SortApplicant(String compareBy) {
+        requireNonNull(compareBy);
+        checkArgument(isValid(compareBy), MESSAGE_CONSTRAINTS);
+
+        switch (compareBy) {
+        case SORTAPPLICANT_BY_DATEAPPLIED:
+            this.compareBy = COMPARE_BY_DATEAPPLIED;
+            this.attribute = ATTRIBUTE_DATEAPPLIED;
+            break;
+        case SORTAPPLICANT_BY_INTERVIEW:
+            this.compareBy = COMPARE_BY_INTERVIEWDATE;
+            this.attribute = ATTRIBUTE_INTERVIEW;
+            break;
+        default:
+            this.compareBy = COMPARE_BY_JOBID;
+            this.attribute = ATTRIBUTE_JOB;
```

Comment:
Consider a case to check for SORTAPPLICANT_BY_ID, then letting default throw an exception? This is also fine but would be better to check that the compare by is actly a case of id.